pkg/insights/insightsclient/requests.go

```diff
@@ -0,0 +1,243 @@
+		return nil, fmt.Errorf("gateway server bad request: %s (request=%s): %s", resp.Request.URL, requestID, string(body))
+	}
+	if resp.StatusCode == http.StatusNotFound {
+		body, _ := io.ReadAll(resp.Body)
+		if len(body) > 1024 {
+			body = body[:1024]
+		}
+		notFoundErr := HttpError{
+			StatusCode: resp.StatusCode,
+			Err:        fmt.Errorf("not found: %s (request=%s): %s", resp.Request.URL, requestID, string(body)),
+		}
+		c.IncrementRecvReportMetric(resp.StatusCode)
+		return nil, notFoundErr
+	}
+
+	if resp.StatusCode >= 300 || resp.StatusCode < 200 {
+		body, _ := io.ReadAll(resp.Body)
+		if len(body) > 1024 {
+			body = body[:1024]
+		}
+		c.IncrementRecvReportMetric(resp.StatusCode)
+		return nil, fmt.Errorf("gateway server reported unexpected error code: %d (request=%s): %s", resp.StatusCode, requestID, string(body))
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		return resp, nil
+	}
+
+	klog.Warningf("Report response status code: %d", resp.StatusCode)
+	return nil, fmt.Errorf("Report response status code: %d", resp.StatusCode)
+}
+
+func (c *Client) RecvSCACerts(ctx context.Context, endpoint string) ([]byte, error) {
+	cv, err := c.getClusterVersion()
+	if err != nil {
+		return nil, err
+	}
+	if cv == nil {
+		return nil, ErrWaitingForVersion
+	}
+	token, err := c.authorizer.Token()
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest(http.MethodPost, endpoint, bytes.NewBuffer([]byte(scaArchPayload)))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	c.client.Transport = clientTransport(c.authorizer)
+	authHeader := fmt.Sprintf("AccessToken %s:%s", cv.Spec.ClusterID, token)
+	req.Header.Set("Authorization", authHeader)
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve SCA certs data from %s: %v", endpoint, err)
+	}
+
+	if resp.StatusCode > 399 || resp.StatusCode < 200 {
+		return nil, ocmErrorMessage(resp)
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			klog.Warningf("Failed to close response body: %v", err)
+		}
+	}()
+
+	return io.ReadAll(resp.Body)
+}
+
+// RecvGatheringRules performs a request to Insights Operator Gathering Conditions Service
+// https://github.com/RedHatInsights/insights-operator-gathering-conditions-service
+// and returns the response body or an error
+func (c *Client) RecvGatheringRules(ctx context.Context, endpoint string) ([]byte, error) {
+	cv, err := c.getClusterVersion()
+	if err != nil {
+		return nil, err
+	}
+	if cv == nil {
+		return nil, ErrWaitingForVersion
+	}
+
+	req, err := c.prepareRequest(ctx, http.MethodGet, endpoint, cv)
+	if err != nil {
+		return nil, err
+	}
+
+	// dynamically set the proxy environment and authentication
+	c.client.Transport = clientTransport(c.authorizer)
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, ocmErrorMessage(resp)
```

Comment:
The `ocmErrorMessage` talks about `OCM API` which can be misleading here.

Reply:
done